File: Examples/Core/Diagram/PredefinedShapes/NBasicShapesExample.cs
using Nevron.Nov.Diagram;
using Nevron.Nov.Diagram.Shapes;
using Nevron.Nov.Dom;
using Nevron.Nov.Graphics;
using Nevron.Nov.Layout;
using Nevron.Nov.UI;

namespace Nevron.Nov.Examples.Diagram
{
    public class NBasicShapesExample : NExampleBase
    {
        #region Constructors

        /// <summary>
        /// Default constructor.
        /// </summary>
        public NBasicShapesExample()
        {
        }

        /// <summary>
        /// Static constructor.
        /// </summary>
        static NBasicShapesExample()
        {
            NBasicShapesExampleSchema = NSchema.Create(typeof(NBasicShapesExample), NExampleBaseSchema);
        }

		#endregion

		#region Example

		protected override NWidget CreateExampleContent()
		{
			// Create a simple drawing
			NDrawingViewWithRibbon drawingViewWithRibbon = new NDrawingViewWithRibbon();
			m_DrawingView = drawingViewWithRibbon.View;

			m_DrawingView.Document.HistoryService.Pause();
			try
			{
				InitDiagram(m_DrawingView.Document);
			}
			finally
			{
				m_DrawingView.Document.HistoryService.Resume();
			}

			return drawingViewWithRibbon;
		}
		protected override NWidget CreateExampleControls()
		{
			return null;
		}
		protected override string GetExampleDescription()
        {
            return @"<p>This example demonstrates the basic shapes, which are created by the NBasicShapeFactory.</p>";
        }

		private void InitDiagram(NDrawingDocument drawingDocument)
		{
			NDrawing drawing = drawingDocument.Content;
			NPage activePage = drawing.ActivePage;

			// Hide grid and ports
			drawing.ScreenVisibility.ShowGrid = false;
			drawing.ScreenVisibility.ShowPorts = false;

			// create all shapes
			NBasicShapeFactory factory = new NBasicShapeFactory();
			factory.DefaultSize = new NSize(120, 90);

			int row = 0, col = 0;
			double cellWidth = 240;
			double cellHeight = 150;

			for (int i = 0; i < factory.ShapeCount; i++, col++)
			{
				NShape shape = factory.CreateShape(i);
				shape.HorizontalPlacement = ENHorizontalPlacement.Center;
				shape.VerticalPlacement = ENVerticalPlacement.Center;
				shape.Text = factory.GetShapeInfo(i).Name;
				shape.MoveTextBlockBelowShape();

				if (i == (int)ENBasicShape.ThreeDBox)
				{
					shape.TextBlock.Padding = new NMargins(0, 15, 0, 0);
				}
				else if (i == (int)ENBasicShape.Concentric)
				{
					shape.TextBlock.Angle = NAngle.Zero;
				}

				activePage.Items.Add(shape);

				if (col >= 4)
				{
					row++;
					col = 0;
				}

				NPoint beginPoint = new NPoint(50 + col * cellWidth, 50 + row * cellHeight);
				if (shape.ShapeType == ENShapeType.Shape1D)
				{
					NPoint endPoint = beginPoint + new NPoint(cellWidth - 50, cellHeight - 50);
					if (i == (int)ENBasicShape.CenterDragCircle)
					{
						beginPoint.Translate(cellWidth / 3, cellHeight / 3);
						endPoint.Translate(-cellWidth / 3, -cellHeight / 3);
					}

					shape.SetBeginPoint(beginPoint);
					shape.SetEndPoint(endPoint);
				}
				else
				{
					shape.SetBounds(beginPoint.X, beginPoint.Y, shape.Width, shape.Height);
				}
			}

			// size page to content
			activePage.Layout.ContentPadding = new NMargins(50);
			activePage.SizeToContent();
		}

		#endregion

		#region Fields

		private NDrawingView m_DrawingView;

		#endregion

		#region Schema

		/// <summary>
		/// Schema associated with NBasicShapesExample.
		/// </summary>
		public static readonly NSchema NBasicShapesExampleSchema;

        #endregion
    }
}

File: Examples/Core/_VB/Diagram/PredefinedShapes/NBasicShapesExample.vb
Imports Nevron.Nov.Diagram
Imports Nevron.Nov.Diagram.Shapes
Imports Nevron.Nov.Dom
Imports Nevron.Nov.Graphics
Imports Nevron.Nov.Layout
Imports Nevron.Nov.UI

Namespace Nevron.Nov.Examples.Diagram
    Public Class NBasicShapesExample
        Inherits NExampleBase
        #Region"Constructors"

        ''' <summary>
        ''' Default constructor.
        ''' </summary>
        Public Sub New()
        End Sub

        ''' <summary>
        ''' Static constructor.
        ''' </summary>
        Shared Sub New()
            Nevron.Nov.Examples.Diagram.NBasicShapesExample.NBasicShapesExampleSchema = Nevron.Nov.Dom.NSchema.Create(GetType(Nevron.Nov.Examples.Diagram.NBasicShapesExample), NExampleBaseSchema)
        End Sub

		#EndRegion

		#Region"Example"

		Protected Overrides Function CreateExampleContent() As Nevron.Nov.UI.NWidget
			' Create a simple drawing
			Dim drawingViewWithRibbon As Nevron.Nov.Diagram.NDrawingViewWithRibbon = New Nevron.Nov.Diagram.NDrawingViewWithRibbon()
            Me.m_DrawingView = drawingViewWithRibbon.View
            Me.m_DrawingView.Document.HistoryService.Pause()

            Try
                Me.InitDiagram(Me.m_DrawingView.Document)
            Finally
                Me.m_DrawingView.Document.HistoryService.[Resume]()
            End Try

            Return drawingViewWithRibbon
        End Function

        Protected Overrides Function CreateExampleControls() As Nevron.Nov.UI.NWidget
            Return Nothing
        End Function

        Protected Overrides Function GetExampleDescription() As String
            Return "<p>This example demonstrates the basic shapes, which are created by the NBasicShapeFactory.</p>"
        End Function

        Private Sub InitDiagram(ByVal drawingDocument As Nevron.Nov.Diagram.NDrawingDocument)
            Dim drawing As Nevron.Nov.Diagram.NDrawing = drawingDocument.Content
            Dim activePage As Nevron.Nov.Diagram.NPage = drawing.ActivePage

			' Hide grid and ports
			drawing.ScreenVisibility.ShowGrid = False
            drawing.ScreenVisibility.ShowPorts = False

			' create all shapes
			Dim factory As Nevron.Nov.Diagram.Shapes.NBasicShapeFactory = New Nevron.Nov.Diagram.Shapes.NBasicShapeFactory()
            factory.DefaultSize = New Nevron.Nov.Graphics.NSize(120, 90)
            Dim row As Integer = 0, col As Integer = 0
            Dim cellWidth As Double = 240
            Dim cellHeight As Double = 150
            Dim i As Integer = 0

            While i < factory.ShapeCount
                Dim shape As Nevron.Nov.Diagram.NShape = factory.CreateShape(i)
                shape.HorizontalPlacement = Nevron.Nov.Layout.ENHorizontalPlacement.Center
                shape.VerticalPlacement = Nevron.Nov.Layout.ENVerticalPlacement.Center
                shape.Text = factory.GetShapeInfo(CInt((i))).Name
                shape.MoveTextBlockBelowShape()

                If i = CInt(Nevron.Nov.Diagram.Shapes.ENBasicShape.ThreeDBox) Then
                    shape.TextBlock.Padding = New Nevron.Nov.Graphics.NMargins(0, 15, 0, 0)
                ElseIf i = CInt(Nevron.Nov.Diagram.Shapes.ENBasicShape.Concentric) Then
                    shape.TextBlock.Angle = Nevron.Nov.NAngle.Zero
                End If

                activePage.Items.Add(shape)

                If col >= 4 Then
                    row += 1
                    col = 0
                End If

                Dim beginPoint As Nevron.Nov.Graphics.NPoint = New Nevron.Nov.Graphics.NPoint(50 + col * cellWidth, 50 + row * cellHeight)

                If shape.ShapeType = Nevron.Nov.Diagram.ENShapeType.Shape1D Then
                    Dim endPoint As Nevron.Nov.Graphics.NPoint = beginPoint + New Nevron.Nov.Graphics.NPoint(cellWidth - 50, cellHeight - 50)

                    If i = CInt(Nevron.Nov.Diagram.Shapes.ENBasicShape.CenterDragCircle) Then
                        beginPoint.Translate(cellWidth / 3, cellHeight / 3)
                        endPoint.Translate(-cellWidth / 3, -cellHeight / 3)
                    End If

                    shape.SetBeginPoint(beginPoint)
                    shape.SetEndPoint(endPoint)
                Else
                    shape.SetBounds(beginPoint.X, beginPoint.Y, shape.Width, shape.Height)
                End If

                i += 1
                col += 1
            End While

			' size page to content
			activePage.Layout.ContentPadding = New Nevron.Nov.Graphics.NMargins(50)
            activePage.SizeToContent()
        End Sub

		#EndRegion

		#Region"Fields"

		Private m_DrawingView As Nevron.Nov.Diagram.NDrawingView

		#EndRegion

		#Region"Schema"

		''' <summary>
		''' Schema associated with NBasicShapesExample.
		''' </summary>
		Public Shared ReadOnly NBasicShapesExampleSchema As Nevron.Nov.Dom.NSchema

        #EndRegion
    End Class
End Namespace
